src/scriptAPI.js

```diff
@@ -0,0 +1,192 @@
+            </states>
+          </stateMachine>
+          <stateMachine name="HelloWorldResponse" id="-343862282">
+            <states>
+              <State name="Done" id="0" />
+            </states>
+          </stateMachine>
+        </stateMachines>
+      </component>
+    </components>
+  </codesConverter>
+</deployment>`;
+}
+     
+
+function getXmlDom(xml) {
+    if (window.DOMParser) {
+        xml = (new DOMParser).parseFromString(xml, 'text/xml');
+    } else if (window.ActiveXObject) {
+        xml = [new ActiveXObject('Microsoft.XMLDOM'), xml];
+        xml[0].async = false;
+        xml[0].loadXML(xml[1]);
+        xml = xml[0];
+    }
+    return xml;
+}
+
+
+function getCodes(componentName, stateMachineName, xmlDom) {
+    var components = xmlDom.getElementsByTagName("component");
+    for (var i = 0; i < components.length; i++) {
+        if (componentName.localeCompare(components[i].getAttributeNode("name").value) == 0) {
+            var componentCode = components[i].getAttributeNode("id").value;
+            var stateMachines = components[i].getElementsByTagName("stateMachine");
+            for (var j = 0; j < stateMachines.length; j++) {
+                if (stateMachineName.localeCompare(stateMachines[j].getAttribute("name").value == 0)) {
+                    return {
+                        "componentCode": components[i].getAttributeNode("id").value,
+                        "stateMachineCode": stateMachines[j].getAttributeNode("id").value
+                    };
+                }
+            }
+            break;
+        }
+    }
+    return null;
+}
+
+
+function getPublish(componentCode, stateMachineCode, xmlDom) {
+    var publishs = xmlDom.getElementsByTagName("publish");
+    for (var i = 0; i < publishs.length; i++) {
+        if (componentCode.localeCompare(publishs[i].getAttribute("componentCode")) == 0 &&
+            stateMachineCode.localeCompare(publishs[i].getAttribute("stateMachineCode")) == 0) {
+            return publishs[i];
+        }
+    }
+    return null;
+}
+
+
+function getEventToSend(componentName, stateMachineName, jsonMessage, xml) {
+    var xmlDom = getXmlDom(xml);
+    var codes = getCodes(componentName, stateMachineName, xmlDom);
+    var publish = getPublish(codes.componentCode, codes.stateMachineCode, xmlDom);
+    var routingKey = publish.getElementsByTagName("topic")[0].textContent;
+
+    var event = {
+        "Header": {
+            "StateMachineCode": { "Case": "Some", "Fields": [parseInt(codes.stateMachineCode)] },
+            "ComponentCode": { "Case": "Some", "Fields": [parseInt(codes.componentCode)] },
+            "EventCode": parseInt(publish.getAttribute("eventCode")),
+            "IncomingType": 0,
+            "MessageType": { "Case": "Some", "Fields": [publish.getAttribute("event")] }
+        },
+        "JsonMessage": JSON.stringify(jsonMessage)
+    };
+    return { event: event, routingKey: routingKey }
+}
+
+
+var Api = function(host, port) {
+
```

Comment:
try to separate the communication part (web socket) and the high level part of the api